nixos/modules/programs/bash/bash.nix

```diff
@@ -82,19 +82,27 @@ in
       promptInit = lib.mkOption {
         default = ''
           # Provide a nice prompt if the terminal supports it.
-          if [ "$TERM" != "dumb" ] || [ -n "$INSIDE_EMACS" ]; then
-            PROMPT_COLOR="1;31m"
+          set_prompt() {
+            if [ "$TERM" == "dumb" ] || [ ! -n "$INSIDE_EMACS" ]; then
+              return
+            fi
+
+            local PROMPT_COLOR="1;31m"
             ((UID)) && PROMPT_COLOR="1;32m"
-            if [ -n "$INSIDE_EMACS" ]; then
-              # Emacs term mode doesn't support xterm title escape sequence (\e]0;)
-              PS1="\n\[\033[$PROMPT_COLOR\][\u@\h:\w]\\$\[\033[0m\] "
-            else
-              PS1="\n\[\033[$PROMPT_COLOR\][\[\e]0;\u@\h: \w\a\]\u@\h:\w]\\$\[\033[0m\] "
+
+            # Emacs term mode doesn't support xterm title escape sequence (\e]0;)
+            if [ ! -n "$INSIDE_EMACS" ]; then
+              local TITLE="\[\e]0;\u@\h: \w\a\]"
             fi
```

Comment:
as the same condition`[ ! -n "$INSIDE_EMACS" ]` is showing up in both these blocks, and it being true results in `return` rightaway;  so, does it mean that the latter block (setting title) will never be run and so can be eliminated?

```
            if [ "$TERM" == "dumb" ] || [ ! -n "$INSIDE_EMACS" ]; then
              return
            fi
```

```
            if [ ! -n "$INSIDE_EMACS" ]; then
              local TITLE="\[\e]0;\u@\h: \w\a\]"
            fi
```